OpenSSL configure step tries to copy files in system directories

Configure step for OpenSSL tries to copy files into system directories. The following error is raised:

```
[  6%] Performing patch step for 'OpenSSL'
-- OpenSSL: Removing 'configure-zlib' log files
-- OpenSSL: OpenSSL_WORKING_DIR: /home/aure/Slicer/build/OpenSSL
-- OpenSSL: cp;/usr/lib/libz.so;/usr/lib/libz.a
-- OpenSSL: Errors detected - See below.

cp: cannot create regular file '/usr/lib/libz.a': Permission denied

CMake Error at /home/aure/Slicer/CMake/ExternalProjectForNonCMakeProject.cmake:104 (message):
  OpenSSL: configure-zlib step failed with exit code '1'.

  Outputs also captured in
  /home/aure/Slicer/build/OpenSSL_configure-zlib_step_output.txt and
  /home/aure/Slicer/build/OpenSSL_configure-zlib_step_error.txt.

  Setting env.  variable EP_EXECUTE_DISABLE_CAPTURE_OUTPUTS to 1 allows to
  disable file capture.

Call Stack (most recent call first):
  /home/aure/Slicer/build/OpenSSL_configure_step.cmake:3 (ExternalProject_Execute)


make[2]: *** [CMakeFiles/OpenSSL.dir/build.make:125: OpenSSL-prefix/src/OpenSSL-stamp/OpenSSL-patch] Error 1
make[1]: *** [CMakeFiles/Makefile2:800: CMakeFiles/OpenSSL.dir/all] Error 2
make: *** [Makefile:114: all] Error 2
```

## Steps to reproduce

```
git clone https://github.com/Slicer/Slicer.git
cd Slicer
mkdir build
cd build
cmake -DSlicer_USE_SYSTEM_curl=ON #This is due to #5049. 
make
```
## Expected behavior

Build process should complete without errors. Some observations:

- At `configure_zlib` step `Slicer_USE_SYSTEM_zlib` is `ON`. I would think that should have been `OFF` if not explicitly configured.
- Maybe  the step executing `-- OpenSSL: cp;/usr/lib/libz.so;/usr/lib/libz.a` is relevant only when `Slicer_USE_SYSTEM_zlib` is `OFF`?

## Environment
- Slicer version: Slicer-4.11.0-2020-07-23
- Operating system: ArchLinux

--- a/SuperBuild/External_OpenSSL.cmake
+++ b/SuperBuild/External_OpenSSL.cmake
@@ -94,6 +94,7 @@ You could either:
 include(\"${Slicer_CMAKE_DIR}/ExternalProjectForNonCMakeProject.cmake\")
 set(CMAKE_BINARY_DIR \"${CMAKE_BINARY_DIR}\")
 set(ENV{VS_UNICODE_OUTPUT} \"\")
+set(Slicer_USE_SYSTEM_zlib ${Slicer_USE_SYSTEM_zlib})
 ")
     if(APPLE)
       file(APPEND ${_env_script}
@@ -124,8 +125,10 @@ set(ENV{VS_UNICODE_OUTPUT} \"\")
     file(WRITE ${_configure_script}
 "include(\"${_env_script}\")
 set(${proj}_WORKING_DIR \"${EP_SOURCE_DIR}\")
+if(NOT Slicer_USE_SYSTEM_zlib)
 ExternalProject_Execute(${proj} \"configure-zlib\" cp ${ZLIB_LIBRARY} ${_zlib_library_dir}/libz.a
   )
+endif()
 ExternalProject_Execute(${proj} \"configure\" sh config --prefix=${EP_SOURCE_DIR} --openssldir=${EP_SOURCE_DIR} --libdir=${EP_SOURCE_DIR} --with-zlib-lib=${_zlib_library_dir} --with-zlib-include=${ZLIB_INCLUDE_DIR} threads zlib shared
   )
 ")

